File: src/frontend/src/types/compliance.ts
export interface ComplianceStandard {
  id: string;
  name: string;
  version: string;
  description: string | null;
}

export interface ComplianceChecklistItem {
  id: string;
  standard_id: string;
  category: string;
  requirement: string;
  description: string | null;
  is_mandatory: boolean;
}

export type ComplianceStatus = "pending" | "compliant" | "non_compliant" | "na";

export interface ProjectComplianceItem {
  id: string;
  project_id: string;
  checklist_item_id: string;
  status: ComplianceStatus;
  evidence_document_id: string | null;
  notes: string | null;
  reviewed_by: string | null;
  reviewed_at: string | null;
  standard_name: string;
  category: string;
  requirement: string;
  is_mandatory: boolean;
}

export interface ComplianceDashboardStandard {
  standard_name: string;
  total_items: number;
  compliant_count: number;
  non_compliant_count: number;
  pending_count: number;
  na_count: number;
  compliance_percentage: number;
}

export interface ComplianceSummary {
  project_id: string;
  standards: ComplianceDashboardStandard[];
  overall_percentage: number;
}

File: src/backend/schemas/compliance.py
import uuid
from datetime import datetime
from enum import StrEnum

from pydantic import BaseModel, ConfigDict


class ComplianceStatus(StrEnum):
    PENDING = "pending"
    COMPLIANT = "compliant"
    NON_COMPLIANT = "non_compliant"
    NA = "na"


class ComplianceStandardRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    name: str
    version: str
    description: str | None


class ComplianceChecklistItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    standard_id: uuid.UUID
    category: str
    requirement: str
    description: str | None
    is_mandatory: bool


class ComplianceChecklistItemCreate(BaseModel):
    standard_id: uuid.UUID
    category: str
    requirement: str
    description: str | None = None
    is_mandatory: bool = True


class ProjectComplianceItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    checklist_item_id: uuid.UUID
    status: ComplianceStatus
    evidence_document_id: uuid.UUID | None
    notes: str | None
    reviewed_by: uuid.UUID | None
    reviewed_at: datetime | None
    created_at: datetime
    updated_at: datetime
    standard_name: str | None = None
    category: str | None = None
    requirement: str | None = None
    is_mandatory: bool | None = None


class ProjectComplianceItemUpdate(BaseModel):
    status: ComplianceStatus | None = None
    evidence_document_id: uuid.UUID | None = None
    notes: str | None = None


class ProjectComplianceItemReview(BaseModel):
    notes: str | None = None


class ComplianceDashboardResponse(BaseModel):
    standard_name: str
    total_items: int
    compliant_count: int
    non_compliant_count: int
    pending_count: int
    na_count: int
    compliance_percentage: float


class ComplianceSummaryResponse(BaseModel):
    project_id: uuid.UUID
    standards: list[ComplianceDashboardResponse]
    overall_percentage: float
